api/grpcserver/mesh_service.go

```diff
@@ -594,7 +595,7 @@ func (s MeshService) MalfeasanceStream(
 	}
 
 	// first serve those already existed locally.
-	if err := s.cdb.IterateMalfeasanceProofs(func(id types.NodeID, mp *types.MalfeasanceProof) error {
+	if err := s.cdb.IterateMalfeasanceProofs(func(id types.NodeID, mp *wire.MalfeasanceProof) error {
```

Comment:
Probably for next PR.

There might also need to be a distinction between the wire-type `MalfeasanceProof` and the domain type of it. Ideally sync/gossip only receive the bytes and call the handler in the `malfeasance` package which itself then calls the handlers for the different types of proofs in their respective packages (via interfaces).

This way `malfeasance` is a dependency of (most) other packages while it itself doesn't depend on anything else.

Reply:
Maybe the `malfeasance` pkg could export a function to create malfeasance proofs of a given type, and with given content so others don't need to depend on `malfeasance/wire`. Something like:

```go
package malfeasance

type ProofKind byte

const (
	MultipleATXs ProofKind = iota + 1
	MultipleBallots
	HareEquivocation
	InvalidPostIndex
)

type ProofContent interface {
	scale.Type
}

type MalfeasanceProof interface {
	... methods needed, probably a way to serialize for publish?
}

func CreateProof(layer types.LayerID, kind ProofKind, content ProofContent) MalfeasanceProof {
	...
}
```

The malfeasance Handler could have a set of validators registered in the form of a map:
```go
type Validator interface {
	Validate(data []byte) (*types.NodeID, ProofContent, error)
}

validators map[ProofKind]Validator
```

Then, the wire-type for, for example, ATX equivocation (`MultipleATXs`) would be in `activation` package along with a validator for it.